### PR TITLE
Add support for setting default agent workspaceVolume

### PR DESCRIPTION
**Pull request by froggabriel**
# What this PR does / why we need it

Previously, there were no values exposed to modify the workspaceVolume type (emptyDir by default) created in the default podTemplate of the default kubernetes cloud. The workspaceVolume values, however, are supported by JCasC and should be exposed by the helm chart.

# Which issue this PR fixes

N/A

# Special notes for your reviewer

N/A

# Checklist

- [X] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] CHANGELOG.md was updated

### CODE REVIEW

**Comment by torstenwalter** [comment]
Did you try to use `agent.yamlTemplate` instead?

**Comment by froggabriel** [comment]
Yes, creating a different type of volume with the name "workspace-volume" (e.g. hostPath) in the agent.yamlTemplate will not override the default emptyDir workspace volume created.

**Comment by froggabriel** [comment]
It is possible to create an entirely separate cloud with a workspaceVolume using `master.JCasC.configScripts`. However, it seems strange to me that `agent.*` will allow all other configurations of the podTemplate to be defined, but not the workspace volume.

**Comment by torstenwalter** [comment]
> yamlTemplate

I thought that one can configure that yaml template takes preference over the configuration options. If I remember correctly then I read somewhere that using the yaml template is actually preferred over configuring things in the settings. One reason could be that it's hard to keep up with the number of options which can be configured for pods.


For me it's also ok if you add the options with this PR. Please add some unit tests in that case.

**Comment by froggabriel** [comment]
Unit tests added. I agree that the yaml template is a better place to specify pod configuration, but setting global defaults for tolerations, node selectors, and workspace volumes through yaml has been problematic. It seems to work a lot better through settings/jcasc.

**Comment by torstenwalter** [comment]
@froblesmartin could you rebase you changes on master and solve the conflicts?

**Comment by froblesmartin** [comment]
> @froblesmartin could you rebase you changes on master and solve the conflicts?

@torstenwalter I think you mean @froggabriel 😅

**Comment by torstenwalter** [comment]
Autompletion... sorry!

File: charts/jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -279,6 +279,27 @@ Returns kubernetes pod template configuration as code
...
+    persistentVolumeClaimWorkspaceVolume:
+    {{- else }}
+    {{ .Values.agent.workspaceVolume.type }}:
+    {{- end }}
+  {{- range $key, $value := .Values.agent.workspaceVolume }}
+    {{- if not (eq $key "type") }}
+      {{ $key }}: {{ if kindIs "string" $value }}{{ tpl $value $ | quote }}{{ else }}{{ $value }}{{ end }}
+    {{- end }}
+  {{- end }}
+{{- end }}
```
Comment (torstenwalter) [inline charts/jenkins/templates/_helpers.tpl:302]: Unit tests are failing I suspect that this adds a newline or something similar.

Comment (froggabriel) [reply charts/jenkins/templates/_helpers.tpl:302]: Yes, there's a newline conflict somewhere, just haven't had a chance to sit down and fix it. Doing now.

**Review by torstenwalter — APPROVED** [review]

**Comment by torstenwalter** [comment]
@froggabriel Thank you!

**Pull request merged**